### DISCUSSION TITLE
Mouse lagging

### DISCUSSION THREAD

**Discussion by DolphinFlopFish** [CosmosOS/Cosmos#2306] [Q&A]
Mouse is lagging after I printed a string to the screen

**Comment by valentinbreiz** [comment]
Hello @DolphinFlopFish can you check if you're used RAM is not increasing?

Do it with GCImplementation.GetUsedRAM()

**Reply by DolphinFlopFish** [reply]
How to I access GCImplementation?

**Reply by DolphinFlopFish** [reply]
Never mind I found it in Cosmos.Core

**Comment by quajak** [comment]
Current accessing the default font will cause a new instance to be created which is slow. Try using a single instance.

**Comment by DolphinFlopFish** [comment]
I will see my RAM

**Comment by DolphinFlopFish** [comment]
The Used RAM is going up

**Comment by DolphinFlopFish** [comment]
The VM freezes when the VM Reaches 227315712 of used memory.
![Screenshot (3)](https://user-images.githubusercontent.com/52783833/174777057-6f986668-51fe-4c75-9805-097eac68274b.png)

**Reply by retrievaaaa** [reply]
That's cause the default VM RAM is set to 256mb, you can increase in the VMware VM settings

**Reply by DolphinFlopFish** [reply]
If I change it it resets back to 256MB

**Reply by retrievaaaa** [reply]
ok, but you should also call Cosmos.Core.Memory.Heap.Collect() every frame or so cause else the used memory will just keep going up

**Reply by DolphinFlopFish** [reply]
OK

**Reply by DolphinFlopFish** [reply]
How do I do that? (Calling it every frame)

**Reply by zarlo** [reply]
Cosmos.Core.Memory.Heap.Collect() is slow dont call it every frame do like ever 100+ frames

**Reply by zarlo** [reply]
1 frame is 1 loop of Run()

**Reply by DolphinFlopFish** [reply]
It doesn't really do anything. (I used a for loop that repeated it 25 times)

for(int i = 0; i < 25; i++)
{
     Cosmos.Core.Memory.Heap.Collect();
}
